eel/eel-string.c

```diff
@@ -304,7 +308,8 @@ eel_str_replace_substring (const char *string,
         {
             break;
         }
-        result_length += replacement_length - substring_length;
+        if (replacement_length > substring_length)
+            result_length += replacement_length - substring_length;
```

Comment:
why only expand?  If the replacement is shorter than the the substring it means the result will be shorter so there's no need to widen the allocation.